### PR TITLE
Allow deployment to app engine

### PR DESCRIPTION
**Pull request by samuelli**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,8 @@
     "build:watch": "npm run generate-gql-types && tsc --watch",
     "clean": "rm -rf lib",
     "check-coverage": "nyc check-coverage --lines 70 --functions 70",
+    "deploy": "rm -rf node_modules && npm install --production && npm run deploy:gcloud && npm install",
```
Comment (aomarks) [inline package.json]: lol

File: src/server/server.ts [hunk 2]
```diff
@@ -190,17 +185,23 @@ app.post('/login', bodyParser.text(), async (req, res) => {
 app.use('/lit-html', express.static('node_modules/lit-html'));
 app.use(express.static('src/server/static'));
 
-// Don't actually start the server when running inside of a test.
-if (!module.parent) {
-  const server = app.listen(8080, 'localhost', () => {
+const environment = process.env.NODE_ENV;
+if (environment !== 'test') {
+  const port = +(process.env.PORT || '') || 8080;
```
Comment (aomarks) [inline src/server/server.ts]: what's the `+` doing?

File: src/server/server.ts [hunk 3]
```diff
@@ -190,17 +185,23 @@ app.post('/login', bodyParser.text(), async (req, res) => {
...
+  let server:Server;
```
Comment (aomarks) [inline src/server/server.ts]: missing space. auto format?

File: src/server/server.ts [hunk 4]
```diff
@@ -190,17 +185,23 @@ app.post('/login', bodyParser.text(), async (req, res) => {
...
+  const callback = () => {
```
Comment (aomarks) [inline src/server/server.ts]: printStatus or something (callback too generic)

Comment (samuelli) [reply src/server/server.ts]: Done

File: src/server/server.ts (hunk 3 above)
Comment (samuelli) [reply src/server/server.ts]: Done

File: src/server/server.ts (hunk 2 above)
Comment (samuelli) [reply src/server/server.ts]: Changed to Number() to be less weird.

File: package.json (hunk 1 above)
Comment (samuelli) [reply package.json]: lol

**Pull request merged**